src/lib/db/clients/mysql.js

```diff
@@ -325,6 +329,27 @@ export async function updateValues(conn, updates) {
   return results
 }
 
+export async function deleteRows(conn, deletes) {
+  const deleteCommands = buildDeleteQueries(knex, deletes)
+
+  const commands = ['START TRANSACTION', ...deleteCommands];
```

Comment:
Inside of here we're using a transaction, but the edits and deletes happen inside separate transactions.

I think it's reasonable for the user to assume they both operate inside a single transaction. Having updates apply, but not row deletes would be pretty confusing I think.